linode/terraform-provider-linode
# Migrate deprecated `SDKVersionString()` to `runtime/debug` in standard library

## 📝 Description

`meta.SDKVersionString()` has been deprecated, and we can use `debug.ReadBuildInfo()` from the standard library `runtime/debug` instead for getting the version of the Terraform SDK.

## ✔️ How to Test

- Enable logging
```bash
export TF_LOG=DEBUG
export TF_LOG_PATH=$PWD/log.txt 
export TF_LOG_PROVIDER_LINODE_REQUESTS=DEBUG
```

- Create some resources: one is with the Framework, and another is with the SDKv2

```terraform
resource "linode_instance" "example" {
  image     = "linode/debian12"
  region    = "us-mia"
  type      = "g6-nanode-1"
  root_pass = "rgvergnbkerognoregnore3vewvrfgbbgefdbnh"
}

resource "linode_token" "name" {
  label = "testtoken"
  scopes = "linodes:read_only"
}
```

- Inspect the `log.txt` file, and ensure both `Terraform-Plugin-Framework/1.5.0` and `Terraform-Plugin-SDK/2.32.0` are in the user agents in the log.

<img width="1162" alt="image" src="https://github.com/linode/terraform-provider-linode/assets/121905282/8447cfb9-4e8c-4147-b320-aff16d1274b6">

<img width="1112" alt="image" src="https://github.com/linode/terraform-provider-linode/assets/121905282/6f2ac032-2d77-4d48-ac5b-4d76f27b01c8">
